### DISCUSSION TITLE
More currency-like?

### DISCUSSION THREAD

**Discussion by tosufever** [dogecoin/dogecoin#3021] [General]
![02](https://user-images.githubusercontent.com/108495285/176808947-7fe80ce5-d7ba-4d73-bcf8-0350e6615735.PNG)
![01](https://user-images.githubusercontent.com/108495285/176809064-1fa46f79-9e89-4fa9-8f85-a0980600848f.PNG)

I'm curious about this.  I've asked some good questions on twitter, but somehow get time traveled back to a 2017 argument that's hard to understand. 

here is what i understand so far:

![04](https://user-images.githubusercontent.com/108495285/176809642-aef0a0ff-be0e-4fb4-a467-8f392b81c1ca.PNG)
![05](https://user-images.githubusercontent.com/108495285/176809803-6dbeaffe-96cd-4c3e-a14e-7773f98e2208.PNG)
![06](https://user-images.githubusercontent.com/108495285/176809892-1e0a15fd-e410-4658-b1f1-36f9337f0a17.PNG)
![07](https://user-images.githubusercontent.com/108495285/176810260-e1b561d7-ca04-4720-a257-cebe54737035.PNG)
![08](https://user-images.githubusercontent.com/108495285/176810336-b4b558d5-fafa-4416-a947-62b0723a652d.PNG)
![01](https://user-images.githubusercontent.com/108495285/176810719-bf04422c-7b99-4b14-b04f-c1af4560448a.PNG)
![02](https://user-images.githubusercontent.com/108495285/176810732-e7c6a11d-5ab6-4bff-a6d0-1d25cbbfca5e.PNG)
![03](https://user-images.githubusercontent.com/108495285/176810609-f087b1f3-2df5-445f-8c72-194d8dd8d188.PNG)

just spitballing here, but what if a 6sec block is more like an optimal system requirement? twitter integration isn't out of scope?

![010](https://user-images.githubusercontent.com/108495285/176812252-07c93a1f-bf14-4567-9724-e665c7210a3b.PNG)
![010a](https://user-images.githubusercontent.com/108495285/176812303-4ee0c0d5-fca2-40a3-b0b6-81516b36002b.PNG)
![011](https://user-images.githubusercontent.com/108495285/176812343-25bbfe1a-b612-49f5-a9bd-94535fba9f7e.PNG)
![0115](https://user-images.githubusercontent.com/108495285/176812888-ef661602-0725-457d-be8b-63f50f711423.PNG)
![015](https://user-images.githubusercontent.com/108495285/176812735-b0a4468f-820b-4913-9e75-60af32ac85f0.PNG)
![12](https://user-images.githubusercontent.com/108495285/176812768-500ab1d3-b92a-45d4-a318-48e3897338c1.PNG)
![013](https://user-images.githubusercontent.com/108495285/176812947-4b958645-d64a-4f51-8d04-dcd3ae7e7871.PNG)
![014](https://user-images.githubusercontent.com/108495285/176812974-bce0beac-2aba-49aa-b530-1ff803fe5f0d.PNG)

kind of a "minimum system requirements" vibe? 

Can anyone help me understand if block time & size is even a relevant topic for an L1 in 2022?  The way i read this; we should slowly be scaling up both block time and size? 

I think it takes a hard fork, and the only time it's happened was in 2014. this is the extent of my knowledge. after some questions & searching, i can't seem to find / understand if we're even considering it?  

![000000A5](https://user-images.githubusercontent.com/108495285/176815234-44933179-8964-4d7f-b7cc-980371f3bca7.PNG)

**Comment by patricklodder** [comment]
I've been exploring options since March last year; testing ideas every time I think of something. To ease your mind, there definitely are options to realize it with a soft fork protocol extension, i.e. how Litecoin did the EB activation.

The reason why I am keeping silent on it is because we really don't get to fail on this, so once announced, we HAVE to deliver, and I truly fear the Twitter amplification machine to mistake ideas for solutions. So I want to have a PoC before I propose what I think is the best way.

**Reply by tosufever** [reply]
First, let me say thank you for all the work put into dogecoin. Much wow!

From your answer, I'm kind of understanding a few things here. It sort of explains why I'm unable to find much info or discussion on the subject, but also leads me to believe that I'm not asking an irrelevant question.

I've asked similar questions on the twitter amplifier, and over on the reddit machine - I'm still confused, but this isn't uncommon. If for some reason this isn't the appropriate venue for this topic/discussion just LMK. I'm least curious about the politics, and most curious about the technical capabilities.

With my limited understanding, I'm assuming a 6sec. 10M block is the unicorn. I hesitate to call it a goalpost? End of the day, with current infrastructure & existing code 6sec might be problematic. It is my understanding there is some threshold when crossed the likelihood of loosing volunteer node runners increases. I've been running a full core node for a little over a year, and for fun monitor resource utilization. I'm not sure I'd notice a difference if core doubled it's resource utilization, but maybe I would if both bandwidth & size did a 10x?

It's cool that some modifications to block time & size can be accomplished with a soft fork. I learn something new every time I ask more questions. I get this is not a simple task. And its understandable to fear the twitter amplifier - they are already running nodes on starlink satellites over there. :-)

"we really don't get to fail on this, so once announced, we HAVE to deliver" this is powerful. I love it.

Maybe without going into details, you could help me conceptualize what an optimal "delivery" might look like? I think I'm getting stuck in one particular rabbit hole. While I'm reading that block time & size should gradually scale over time, very few modifications to these metrics ever happen (on any blockchain). I'm assuming because of immediate need, risks & political economic factors. Seems this has resulted in a very reactionary approach to modify block time or size. (it's not currently scaling over time).

Lets assume we set the scale somewhere between (1min & 6sec) and (1Mb & 10Mb). Do you throw a hail mary & make a single major change, or are we looking at 10 smaller moves over time?

[Hypothetically] lets say some adjustment to time or size [must] get rolled out with the next update. how much of a change/delta to time and size could you comfortably make? 1% 10% 2X?

I guess I'm asking what is the realistic scale to consider given the existing node infrastructure? Can we 2x (30sec + 2Mb) and cause no disruptions, (but maybe 15sec +4Mb) starts to become problematic in one way or another?

*edit - moved to reply. clicked on the wrong box.

**Reply by patricklodder** [reply]
To answer your last question first: filling current capacity will be problematic for what I currently expect to be half the network, but hard data on node deployment capacity is very scarce so I'm honestly hoping to be pleasantly surprised on that. Let's put it at 40-60%, and note that this is on top of the ~ 60% we already lost over the course of the past year.

I'm **explicitly not** looking to simply change 2 parameters, propose a soft-wrapped hard fork and that's it, because then we will have to pray the network doesn't implode into 4 payment processors, 3 minings pools and a supercomputer running it. This is Dogecoin, not Solana. But my main reason is that it is in my opinion too much risk for a one-time change that will then need to be done over and over every time someone disrupts something and we need more capacity. Removing the limits completely is just going to attract nonsense tx that have no financial meaning whatsoever and will just make the chain meaningless and bloated; we're a crypto**currency** as the title of your topic suggests. (more currency-like is not a thing btw, we already are a currency, no "like" suffix is needed.)

What I am looking into are structural solutions that enable both immediate and future solutions - I don't want to address the surface problem, I want to address the root cause, and do it in the most liberal, freedom-preserving way I can. My self-imposed requirements during the thought process are:

- Optionality. We need to have as little impact as possible on nodes that just want to validate the basic chain and make sure we can still run a fully validating node on a raspberry pi (in pruned mode.)
- Permissionlessness, within a hard consensus framework - like we have now. It is important that we enable, not dictate, and make good alternatives equitable without forcing ideas to hard-fork.
- Integrated. Needs to be part of the "L1" chain, even when we're working on say, optional EBs - no L2/3/69 as a means to scale the L1.
- Transparency. We need to have an audit trail. Even if this is proof of a ZK rollup, the proof still needs to be there and we need to make sure that every node that wants to audit, can audit. "Don't trust, verify" is an important principle to maintain when we're going to deal with ever-increasing commercial interests.
- Security: We need to maintain sovereign security at any time. No solutions that require delegations to miner decisions that cannot be verified on-chain. 

At the risk of creating false hopium, here's a **fictional scenario** from where we are today:

1. Enable [versionbits](https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki) and do the process with [relative locks](https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki) because that's the only remaining non-contentious protocol extension from the plan I published last year.
2. Add Litecoin [extension blocks](https://github.com/litecoin-project/lips/blob/master/lip-0002.mediawiki) through another versionbits upgrade - preferably with some lessons learned / improvements.
3. Add [mina-protocol style rollups](https://docs.minaprotocol.com/static/pdf/technicalWhitepaper.pdf) to be a first EB (note: we may want to go for something that is not completely out of this world ambitious as a first EB instead, but let's do it extreme in theory right now)

Now suddenly we have full ZK rollups on an EB protocol extension on our L1, where nodes that don't operate on the EB, do not need to validate anything more than a small ZK proof.

These are the kind of scenario's I'm playing with right now and the research I'm doing (example: how can we depend on ouroboros fast consensus with a "slow" beacon chain that timestamps extension blocks.)

Hope this helps.

**Reply by GiverofMemory** [reply]
> I've been exploring options since March last year; testing ideas every time I think of something. To ease your mind, there definitely are options to realize it with a soft fork protocol extension, i.e. how Litecoin did the EB activation.
> 
> The reason why I am keeping silent on it is because we really don't get to fail on this, so once announced, we HAVE to deliver, and I truly fear the Twitter amplification machine to mistake ideas for solutions. So I want to have a PoC before I propose what I think is the best way.

I think bitcoin's biggest mistake was trying to only do soft forks.  I think a basic "change the parameter" method will save coders headaches in the future.  Hard fork is no problem all users and miners will want faster and bigger blocks.  I really don't think we need to nor should overcomplicate a simple change like this.  The root cause of everything is Satoshi ninja hard forking bitcoin to include a blocksize limit due to Hal Finney FUD.  Bitcoin SV has many gigabyte blocksize limit and very tiny fees (much smaller than ours) and the blocks almost always have mountains of space left over.  Fees are what keep the blockchain size in check, not an arbitrary blocksize limit that has never done anything positive in the industry to a coin that has reasonable fees.

**Reply by GiverofMemory** [reply]
> First, let me say thank you for all the work put into dogecoin. Much wow!
> 
> From your answer, I'm kind of understanding a few things here. It sort of explains why I'm unable to find much info or discussion on the subject, but also leads me to believe that I'm not asking an irrelevant question.
> 
> I've asked similar questions on the twitter amplifier, and over on the reddit machine - I'm still confused, but this isn't uncommon. If for some reason this isn't the appropriate venue for this topic/discussion just LMK. I'm least curious about the politics, and most curious about the technical capabilities.
> 
> With my limited understanding, I'm assuming a 6sec. 10M block is the unicorn. I hesitate to call it a goalpost? End of the day, with current infrastructure & existing code 6sec might be problematic. It is my understanding there is some threshold when crossed the likelihood of loosing volunteer node runners increases. I've been running a full core node for a little over a year, and for fun monitor resource utilization. I'm not sure I'd notice a difference if core doubled it's resource utilization, but maybe I would if both bandwidth & size did a 10x?
> 
> It's cool that some modifications to block time & size can be accomplished with a soft fork. I learn something new every time I ask more questions. I get this is not a simple task. And its understandable to fear the twitter amplifier - they are already running nodes on starlink satellites over there. :-)
> 
> "we really don't get to fail on this, so once announced, we HAVE to deliver" this is powerful. I love it.
> 
> Maybe without going into details, you could help me conceptualize what an optimal "delivery" might look like? I think I'm getting stuck in one particular rabbit hole. While I'm reading that block time & size should gradually scale over time, very few modifications to these metrics ever happen (on any blockchain). I'm assuming because of immediate need, risks & political economic factors. Seems this has resulted in a very reactionary approach to modify block time or size. (it's not currently scaling over time).
> 
> Lets assume we set the scale somewhere between (1min & 6sec) and (1Mb & 10Mb). Do you throw a hail mary & make a single major change, or are we looking at 10 smaller moves over time?
> 
> [Hypothetically] lets say some adjustment to time or size [must] get rolled out with the next update. how much of a change/delta to time and size could you comfortably make? 1% 10% 2X?
> 
> I guess I'm asking what is the realistic scale to consider given the existing node infrastructure? Can we 2x (30sec + 2Mb) and cause no disruptions, (but maybe 15sec +4Mb) starts to become problematic in one way or another?
> 
> *edit - moved to reply. clicked on the wrong box.

First what you have to realize is that our blocks are not even full currently, so making this change will have no effect currently.  But it will allow the network to scale as needed, and remember harddrive space scales over time much like moore's law and I believe even more reliably since we are not close to the atomic scale bottleneck like moore's law is.  So all this to say that yes node operators can and should be willing to upgrade harddrives every few years...

**Reply by GiverofMemory** [reply]
> * Optionality. We need to have as little impact as possible on nodes that just want to validate the basic chain and make sure we can still run a fully validating node on a raspberry pi (in pruned mode.)

First we are running a global payments network so a raspb pi I don't think should be any sort of benchmark.  But for light verifying nodes how about instead of trying to keep up with the 6 second verification time, just make the light wallet only ping the network every minute (10 blocks) and then verify everything looks ok?  I don't think we need a raspb to be verifying every announced transaction in real time...

**Reply by patricklodder** [reply]
> First we are running a global payments network so a raspb pi I don't think should be any sort of benchmark. 

Keyword: we. My rationale is that if you break a lower end requirement once, then there's no going back. Note that this is a personal thought process constraint - like the other ones in that list - and not an implementation constraint, for the simple reason that if I don't try, then it will be the first thing that gets sacrificed. So it truly is a benchmark because it's currently the entry-level full node and shibes are actually building on this platform - I feel it would be a disservice to them to just rid ourselves of the platform as a viable thing by not considering supporting it. Besides, I have for years ran a bitcoind and lnd together on an odroid HC1. It works beautifully and means that the next 10x-50x (our average utilized block size sits at 10-12 kilobytes) should still support that platform.

> just make the light wallet only ping the network every minute

This makes the network client-server more than peer-to-peer, through service provision. We're already dealing with node classes ever since the introduction of Multidoge in 2014 as the first SPV client.

- [BIP-37](https://github.com/bitcoin/bips/blob/master/bip-0037.mediawiki) SPV clients only work when there are nodes that provide the `NODE_BLOOM` service - see #2733 why that can lead to issues and may ultimately be undesirable from a censorship resistance perspective. Also see [BIP-157](https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki) for the next gen implementation that is reducing the exposure for service provider nodes, but still retains the concept of service provision.
- Pruning nodes only function when there are full nodes (i.e. providing the `NODE_NETWORK` service) connected to them, because there is always the risk of a fork larger than the blocks we kept, or downtime past the guaranteed buffer in general.

Segmentation into these node classes makes part of the network (the part that relies on a specific subset of services) easier to attack. Since we're talking about the core infrastructure here we have to be extremely careful to not introduce weaknesses that decrease the effective protection for large numbers of wallets. Example: if 1 million shibes use a wallet that ultimately depends on *n* nodes then all we need to do as an attacker is take those *n* nodes down. The higher the requirements are for the network to function, the lower *n* will be, the easier it will be to attack.

> light wallet

How does any light wallet implementation protect against double spends of an input to its incoming transactions? How does it even detect that that happened?

> Bitcoin SV has many gigabyte blocksize limit and very tiny fees 

How many B-SV nodes do you run? I will take down a Dogecoin node of mine and run an B-SV node. I'll let you know what I find.

> Ethereum is around a 12 second blocktime if I am remembering correctly and doesn't have many issues.

How many Ethereum nodes do you run?

**Reply by tosufever** [reply]
i hate to beat a dead horse & reopen this thread, but i'm still confused. 

first, i want to say much respect to this venue, and much respect to all the responses to my probably redundant questions. much wow. every one here deserves much respect!

honestly, i'm still not sure i'm even asking a relevant question. 

at first glance this probably looks like a fanboy thread. lv1 - i'm just starting to geek out around here. 
and it's just one narrative (of many) that has me crazy curious. curious enough to endure this current eye-roll you just made.

i want to make sure we are specific... i am asking about scaling L1 block time (and block size).
this is a 2022 question, with a 2017 narrative, and the same two specific metrics. 

my brain correlates [block time] to something like ethernet standards. 10/100/1000 - the pace the network runs. how fast it goes. period. in “blocktime” these variables for dogecoin were hard set in 2013 and have never changed.

mostly we redirect to [protocol improvements] my brain compares to maybe [tcp/ip] - the code that runs on the network at the speed negotiated above.  

obviously if you double the speed of the network, the protocol (and in our case - node infrastructure) would still have to support it.  

consider there is often no visible benefit to upgrade some networks from 1gb to 10gb

depending on the objective (this maybe where i'm getting tripped up) a protocol improvement may already exist with a better solution to meet performance requirements. more speed much wow.  

the pictorial narrative discusses scaling block time and size (over time) and correlate with improvements in computing power. dogecoin has never modified either metric. and it has been around awhile. :)  

my question is... should we? (shouldn't we?) can we? is this even relevant? increase the block time and block size?

while the spotlight is on POS, bright minds are discussing need for the fastest, cheapest, and still decentralized layer 1. going as far as to say that doge should (could) be faster (already). here are some recommended specs… (i’m reading it as minimum system requirements?) 

to be fair i don't understand the current scalability, but somehow it seems important?

hypothetically, is there a least disruptive (somehow doesn't change the economics) (minimal code hours) (spaceX not NASA method) - way to: at the next update grab an extra 5% block time & block size? simply to understand how the network responds? this way should the need arise to scale the network in this direction, we have some better idea about scalability and capabilities? i'm just looking at this like it should be a tool in the tool belt? 

it seems like there is apprehension / we are afraid / don't want / don't need to / it's just logically impossible to scale / don’t really even like to discuss / maybe i'm just asking the wrong question – talk specifically about block time.

the first time i asked, there was just some talk of birds. a potential event to integrate with a large mainstream platform [town hall] that could result in a more rapid capacity shift or onset of new users. inversely maybe there is some minimum system requirement. the scale here is fairly significant. to do it right the platform doesn't become a defacto l2, all tx are onchain. 

today radioDoge is obsolete. there is enough eWaste in the world to give every breathing human a functioning (android) device with basic PCS data capabilities... all the last gen cell phones we churn out... working on every inch of the planet. this is backwards. usually only new devices get enhanced communication abilities. again another possible instance where inbound new users doesn't happen gradually. 

satellite to PCS data is slow. it's not connecting the planet to youtube. but it’s fast enough to support planet wide edge blockchain transactions out of the gate. the scale here is exponentially larger. the expectation was to shrink the sat receiver to fit in the cell phone, not use all the old phones to give everyone in the world basic connectivity. this is giga scale and it's coming giga faster than anyone expects.

is there any alignment?

**Reply by patricklodder** [reply]
I'm sorry I failed to be clear and look scared (?!?)

Here's the code:

```python
ideas[elon] = 10 * 10 * 1M = 100M
ideas[vlad] = 1G * 60 = 60G
ideas[bsv_peeps] = 2G

desired_capacity = sum(ideas)

current_capacity = 1M
desired_increase = desired_capacity / current_capacity # = 62.1G / 1M = 62100

current_utilization = 1M / 100
utilization_increase = desired_increase / current_utilization # 62100 / 0.01 = 6,210,000
```

That's just 3 ideas requiring a 6,210,000x on current utilization. We can tweet all we want but that is simply not going to happen with current tech because this is not what it is designed to do. Therefore, in my opinion it is not a question of parametrization, because that'd be a waste of time - it won't ever stop: the 100x after the first 100x is a 10,000x against the original, and the next disruption is a 1,000,000x. If you want to enable radical disruption, you have to make radical changes. I'm not afraid, I just don't want to spend my time on stuff that is not going to help enable what is desired, whilst completely killing fought-for decentralization and censorship resistance as we go, and then do that over and over and over. The tweets you mention are a one-sided desire/idea and do not represent Dogecoin as a platform.

I gave a possible solution above that I think deals with this in a future-proof fashion. It included enabling parallel soft fork proposals, so that better ideas than mine can compete and win.

Let's not make devs or influencers all-powerful by doing series of hard forks that will ultimately destroy sovereignty; you cannot have "the people's coin" if we don't protect the bottom line of censor-proof money, and for that we need decentralization. Operating expense is not just fees, it's also the cost to run a node, so if we make the fees 100x lower and then increase the cost to run a node 100x, we're back where we started, except with less decentralization.

Working around expenses through centralization (i.e. how Ethereum and every single imitator of that does it) is not great in the face of increasingly audacious governments, but if you don't care and want that, just use Solana. Maybe they'll be open to having a good bridge - I hear they're in need some TVL that actually exists.

**Reply by tosufever** [reply]
thank you for the insight. it's always appreciated. you'll have to forgive how my brain works; i'm not suggesting any of this should be done, or that anyone is really "scared". i was just sort of joking that the more specific i drill down and ask questions (to anyone) about block time... somehow it feels like there is an immediate offensive or defensive side to the conversation. 

i still feel like i'm not asking the right questions. because here i am typing again. 

"The tweets you mention are a one-sided desire/idea and do not represent Dogecoin as a platform" - i respect that. maybe this is the answer i'm not hearing. the last thing i want to see, is any degradation of the decentralization. doge is probably the last real hope crypto has in that department. so if there is/was a choice (one or the other?) around this narrative it's a no brainier. 

from my perspective it looks like like a tantalizing, rare, and unique opportunity. planet wide sat to cell coverage (basically tomorrow). charles can finally stop speculating about blockchain adoption in rural africa. (again sorry radioDoge). what sticks out to me is the fact dogecoin is already positioned in these seemingly unrelated opportunities front & center. i assume a new chain - designed to fit - would be the easiest solution. yet there is a silver platter (can I call it an “offer”?) on the table which would most likely integrate it into a platform that is accessible to all shibs everywhere on earth. the carrier "app" could function with connectivity even if the device isn't signed up for service.  ironically it’s literally all the tools necessary to make “the peoples currency” a reality.

it seems significant enough to consider making some compromises if necessary, but not black and white enough to send everything down a dark-sided death spiral.  just to reiterate, if the promise comes at the complete expense of decentralization this conversation was irrelevant before it started.  

i keep thinking maybe the vitalic elon conversation was not about changing dogecoin, rather an ideal set of requirements to directly integrate [a layer one blockchain] directly into an arbitrary [platform], ecosystem, or town hall - WITHOUT it digressing into a defacto layer 2. "if we could go back and do it different" or "here is a roadmap"

maybe doge fits here, maybe it doesn't - i don't know.  

it's possible i'm derailing the conversation when i use the term “scale” and we are intermittently speaking about capacity. i'm not looking for disruption here, and certainly not changes that result in anything "orders of magnitude" different. maybe it's not a capacity issue at all, maybe it is just time?

what if block time (and size) only need to be modified once to meet some minimum system requirements? after all they were arbitrarily set in 2013 without further modification. i'm inclined to believe, based on the entire context of the elon vitalik conversation that maybe their "ideal" block time is already achievable on the dogecoin network without killing all the things that makes it doge. it was kind of the theme of the interaction. otherwise why would there even be a discussion? a new chain is easy enough to create.  

from a different perspective: maybe at [1 minute] block time - the most basic "platform" integration features  [on layer one] aren’t "fast enough" for instant gratification of its users. think wallet as your identity & credentials. 

However if you started with a blockchain that runs at a faster time "X" per block. everything just works & the user experience completely changes. (they were fairly specific about 7 seconds) maybe it's a "magic threshold" where base layer transactions are just "fast enough" to provide the necessary instant gratification of its core platform blockchain integration features, but just "SLOW enough" that it would run on dogecore today (without compromising the protocol, economics, or it’s identity) 

and since we're in the simulation, someone already did some math. it almost feels like 7 seconds is probably the sweet spot where everything just works (existing infrastructure & protocol) - planet wide - at PCS radio latency.

**Reply by patricklodder** [reply]
Everything Dogecoin was arbitrarily set in 2013, except those things that had to change because they were threats to the survival of the protocol: remove pseudorandom rewards to eliminate sniping, enforced a fee and dust policy to prevent spam, accept auxpow to prevent 51% attacks. Everything else is either there because it's the dumbest possible thing to do, or it is there by accident.

The answer to a hard fork is always no, unless there is no other way. I think I proposed another way. I see zero arguments against my counter proposal so what is your objection? Why are you so set on the 7 seconds? What use-case is it for if not scaling?

**Comment by GiverofMemory** [comment]
I fully agree with elon and billy on this, Ethereum is around a 12 second blocktime if I am remembering correctly and doesn't have many issues.
